modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -152,8 +153,9 @@ private List<String> findRoutingPaths(String indexName, Settings allSettings, Li
             .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, dummyShards)
             .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, shardReplicas)
             .put(IndexMetadata.SETTING_INDEX_UUID, UUIDs.randomBase64UUID())
+            .put(IndexSettings.MODE.getKey(), IndexMode.TIME_SERIES)
```

Comment:
Can we avoid this change? What is the reason tsdb needs to be enabled here? I think we just do some mapping stuff that doesn't require tsdb to be enabled? I could be wrong.

Reply:
There may be more settings causing loading problems below, e.g. any of these can fire:
https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/IndexMode.java#L54

It seems safer to load this as TSDB instead of removing all these settings.

Reply:
Sure this work but why is it preferable to fake it that we are not using a time-series mode, instead of getting something closer to the original mapping? We're only faking the routing path because it's the one we're trying to retrieve in this function. This also feels more fragile - each time we add something to TSDB mode validation we'll have to update this function too..

Reply:
I think faking time series mode can be as fragile? If would add a new time series index setting or update the time series index settings validation, then it is likely we would need to make changes here too.

I think removing an index setting here is easier then fixing the logic that fakes time series mode. And for index routing generation the `MapperService` doesn't require the dummy index to be in tsdb mode. So I prefer to that this way.

Note that these settings are not added by the `DataStreamIndexSettingsProvider`, but originate from the template. The start and end time settings shouldn't be added to template to begin with (although we do this sometimes in tests to make the test easier).

Reply:
This function is called only when `IndexMode` is `TIME_SERIES`, see the function above. This means that we never fake a time-series mode; I can remove this line and nothing will change, but I think it's good to have it here for visibility. If there are other issues with the mapping, it will fail validation either way. Here, we're just trying to fix routing path.

This is broken as is right now. I got an error here when I tried to use path match in a template with start_time. This is very hard to spot, of course - and it would never be an issue if we didn't change the mode here to non-tsdb.

Reply:
> This is broken as is right now. I got an error here when I tried to use path match in a template with start_time. This is very hard to spot, of course - and it would never be an issue if we didn't change the mode here to non-tsdb.

True, because we don't remove the start and end time settings?


>  This function is called only when IndexMode is TIME_SERIES, see the function above.

This is true. But at this point in time we don't know the routing paths, so setting up a temporary indexService with index.mode=time_series would require supplying dummy routing values. This temporary IndexService doesn't have to be in time series mode, we just need the MapperService which keyword fields have the time series dimension attribute set.

Reply:
We've resolved this :) 

Both our approaches workaround the fact that we can't setup time series correctly for the temporarily index/mapper service, because routing path is unknown at this point. Let's keep this approach where we use a fake dimension, to ensure that both this temporarily index/mapper service and subsequent temporarily index/mapper service that are really used for validating are more consistent with each other.

Reply:
Agreed, thanks for bearing with me here :)